File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.lower()
        return self.check(s, 0, len(s)-1)

    def check(self, s: str, start : int, end : int) -> bool:
        if(start>=end):
            return True
        if(not s[start].isalnum()):
            return self.check(s, start+1, end)
        if(not s[end].isalnum()):
            return self.check(s, start, end-1)
        if(s[start] is not s[end]):
            return False
        return self.check(s, start+1, end-1)

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        transform(s.begin(), s.end(), s.begin(), ::tolower);
        return check(s, 0, s.length()-1);
    }
    bool check(string s, int start, int end){
        while(start<end){
            if(!isalnum(s[start])){
                start+=1;
                continue;
            }
            if(!isalnum(s[end])){
                end-=1;
                continue;
            }
            if(s[start] != s[end]) return false;
            start++;
            end--;
        }
        return true;
    }
};